### PR TITLE
[feature](planner) multi partition create by integer column

### PR DESCRIPTION
**Pull request by catpineapple**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Create partitions use ：
```
PARTITION BY RANGE(integer_col)(
        FROM (10) TO (1000) INTERVAL 50
)
```
## Checklist(Required)

* [x] Does it affect the original behavior
* [x] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by catpineapple** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.56 seconds
 stream load tsv:          429 seconds loaded 74807831229 Bytes, about 166 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          60 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230512120732_clickbench_pr_142942.html

**Comment by catpineapple** [comment]
run buildall

File: regression-test/suites/partition_p0/multi_partition/test_multi_partition.groovy [hunk 1]
```diff
@@ -321,4 +321,37 @@ suite("test_multi_partition") {
...
+            k12 datev2 NOT NULL,
+            k13 datetimev2 NOT NULL,
+            k14 datetimev2(3) NOT NULL,
+            k15 datetimev2(6) NOT NULL,
+            k7 varchar(20) NOT NULL,
+            k8 double max NOT NULL,
+            k9 float sum NOT NULL )
+        AGGREGATE KEY(k1,k2,k3,k4,k5,k6,k10,k11,k12,k13,k14,k15,k7)
+        PARTITION BY RANGE(k3) (
+            FROM (1000) TO (10000) INTERVAL 200
```
Comment (morningman) [inline regression-test/suites/partition_p0/multi_partition/test_multi_partition.groovy:345]: What if `FROM (1000) TO (1001) INTERVAL 200`?

Comment (catpineapple) [reply regression-test/suites/partition_p0/multi_partition/test_multi_partition.groovy:345]: build one partition : [1000,1001)。
```
            PartitionValue lowerPartitionValue = new PartitionValue(beginNum);
            beginNum += interval;
            Long currentEnd = Math.min(beginNum, endNum);
            PartitionValue upperPartitionValue = new PartitionValue(currentEnd);
```

**Comment by catpineapple** [comment]
run buildall

**Comment by catpineapple** [comment]
run buildall

**Comment by catpineapple** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by morningman — CHANGES_REQUESTED** [review]
Please update the document, and add more test cases, such as the case I gave.

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.75 seconds
 stream load tsv:          446 seconds loaded 74807831229 Bytes, about 159 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          25.5 seconds inserted 10000000 Rows, about 392K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230528142246_clickbench_pr_152260.html

**Comment by catpineapple** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.03 seconds
 stream load tsv:          447 seconds loaded 74807831229 Bytes, about 159 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          30.6 seconds inserted 10000000 Rows, about 326K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230529033325_clickbench_pr_152455.html

**Comment by catpineapple** [comment]
run buildall

**Comment by catpineapple** [comment]
run buildall

**Comment by catpineapple** [comment]
run buildall

**Comment by catpineapple** [comment]
run buildall

**Comment by catpineapple** [comment]
run buildall

**Comment by catpineapple** [comment]
run p0

**Review by JNSimba — APPROVED** [review]
LGTM

**Comment by catpineapple** [comment]
run buildall

**Comment by catpineapple** [comment]
run p0

**Comment by catpineapple** [comment]
ing update fil

**Comment by catpineapple** [comment]
run buildall

**Comment by catpineapple** [comment]
run buildall

**Comment by catpineapple** [comment]
run buildall

**Comment by catpineapple** [comment]
run buildall

**Comment by catpineapple** [comment]
run buildall

**Review by zy-kkk — APPROVED** [review]
LGTM

**Comment by catpineapple** [comment]
run p1

**Pull request merged**